server/graphql/mutations/orders.js

```diff
@@ -311,63 +309,71 @@ export function updateSubscription(remoteUser, args) {
     ]
   };
 
-  return models.Order.findOne(query)
-  .tap(order => {
-    if (!order) {
-      throw new Error("Subscription not found")
-    }
-    return Promise.resolve();
-  })
-  .tap(order => {
-    if (!remoteUser.isAdmin(order.FromCollectiveId)) {
-      throw new errors.Unauthorized({
-        message: "You don't have permission to update this subscription"
-      })
-    }
-    return Promise.resolve();
-  })
-  .tap(order => {
-    if (!order.Subscription.isActive) {
-      throw new Error('Subscription must be active to be updated');
-    }
-    return Promise.resolve();
-  })
-  .tap(order => {
+  let order = await models.Order.findOne(query);
 
-    let updatePromise;
-    let newPm;
-    
-    // TODO: Would be even better if we could charge you here directly
-    // before letting you proceed
-
-    // means it's an existing paymentMethod
-    if (paymentMethod.uuid && paymentMethod.uuid.length === 36) {
-      updatePromise = models.PaymentMethod.findOne({where: { uuid: paymentMethod.uuid}})
-        .then(pm => {
-          if (!pm){
-            throw new Error('Payment method not found with this uuid', paymentMethod.uuid);
-          }
-          newPm = pm;
-        })
-    } else {
-      // means it's a new paymentMethod
-      const newPMData = Object.assign(paymentMethod, { CollectiveId: order.FromCollectiveId })
-      updatePromise = models.PaymentMethod.createFromStripeSourceToken(newPMData)
-        .then(pm => newPm = pm)
-    }
+  if (!order) {
+    throw new Error("Subscription not found")
+  }
+  if (!remoteUser.isAdmin(order.FromCollectiveId)) {
+    throw new errors.Unauthorized({
+      message: "You don't have permission to update this subscription"
+    })
+  }
+  if (!order.Subscription.isActive) {
+    throw new Error('Subscription must be active to be updated');
+  }
 
-    // determine if this order was pastdue
-    if (order.Subscription.chargeRetryCount > 0) {
-      const updatedDates = getNextChargeAndPeriodStartDates('updated', order);
-      const chargeRetryCount = getChargeRetryCount('updated', order);
+  if (amount) {
```

Comment:
Need to check that `amount` has a minimum. The order creation flow has an amount - I think `50` or `100` cents. Otherwise, people maybe able to set really low values that will be problematic with Stripe.

Reply:
@asood123 On the backend side, I haven't found the minimal amount handling anywhere. Any pointer? On the frontend side, I only see some 'min' that are matching the Tier.

Reply:
It was in our rest end point and looks like disappeared in the graphql, which is checking only for amount > 0. I'd suggest putting a min of 100 cents (regardless of currency). It's not perfect, but good enough to prevent users from setting too low amounts that will mess with Stripe and our system.

Reply:
Scratch that, saw you already put that in place. :)